src/pages/home/report/ReportActionCompose/AttachmentPickerWithMenuItems.tsx

```diff
@@ -168,10 +170,58 @@ function AttachmentPickerWithMenuItems({
             },
         };
 
-        return ReportUtils.temporary_getMoneyRequestOptions(report, policy, reportParticipantIDs ?? []).map((option) => ({
+        let list: PopoverMenuItem[] = ReportUtils.temporary_getMoneyRequestOptions(report, policy, reportParticipantIDs ?? []).map((option) => ({
             ...options[option],
         }));
-    }, [translate, report, policy, reportParticipantIDs, isDelegateAccessRestricted]);
+
+        if (canUseCombinedTrackSubmit) {
+            const trackOnlyCreateExpense = {
+                icon: getIconForAction(CONST.IOU.TYPE.CREATE),
+                text: translate('iou.createExpense'),
+                onSelected: () => selectOption(() => IOU.startMoneyRequest(CONST.IOU.TYPE.TRACK, report?.reportID ?? '-1'), true),
+            };
+
+            const submitOnlyCreateExpense = {
+                icon: getIconForAction(CONST.IOU.TYPE.CREATE),
+                text: translate('iou.createExpense'),
+                onSelected: () => selectOption(() => IOU.startMoneyRequest(CONST.IOU.TYPE.SUBMIT, report?.reportID ?? '-1'), true),
+            };
+
+            if (list.some((option) => options[CONST.IOU.TYPE.TRACK].text === option.text) && list.some((option) => options[CONST.IOU.TYPE.SUBMIT].text === option.text)) {
+                list = list.reduce((acc, item) => {
+                    if (item.text === translate('iou.submitExpense')) {
+                        acc.push(submitOnlyCreateExpense);
+                    } else if (item.text !== translate('iou.trackExpense')) {
+                        acc.push(item);
+                    }
+
+                    return acc;
+                }, [] as PopoverMenuItem[]);
+            } else if (list.some((option) => options[CONST.IOU.TYPE.TRACK].text === option.text)) {
+                list = list.reduce((acc, item) => {
+                    if (item.text === translate('iou.trackExpense')) {
+                        acc.push(trackOnlyCreateExpense);
+                    } else {
+                        acc.push(item);
+                    }
+
+                    return acc;
+                }, [] as PopoverMenuItem[]);
```

Comment:
Would it be easier to not add the other options to the list in the first place?

Reply:
Updated to the approach without new elements